### PR TITLE
Create incomplete draft of inadequate privacy controls

### PR DESCRIPTION
**Pull request by slortz**

### CODE REVIEW

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+# Threat Agents
+
+**Application Specific**
+
+Privacy controls are concerned with protecting Personally Identifiable Information (PII). This information is valuable to attackers for various reasons, e.g., for
+
+- impersonating the victims in cases of fraud,
+- misusing the victims' payment data, or
```
Comment (kunwaratulhax0r) [inline 2023-risks/m4-inadequate-privacy-controls.md:13]: - misusing the victim's payment data, or

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- blackmailing the victims with sensitive information.
```
Comment (kunwaratulhax0r) [inline 2023-risks/m4-inadequate-privacy-controls.md:14]: - impersonating the victims in cases of fraud,
- misusing the victim's payment data, or
- blackmailing the victim's with sensitive information.
- Absence of data deletion mechanisms
- Limited user awareness and control
- Absence of granular privacy settings

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+# Attack Vectors	
+
+**Exploitability MEDIUM**
+
+Typical sources for PII are either protected or hard to access, e.g., the sandbox of the app, the network communication with the server, the app's logs and backups. Some are easier to observe, like URL query parameters and clipboard content.
+
+Obtaining PII thus requires the attacker to breach security on another level. He could eavesdrop on the network communication, access file system, clipboard or logs with a trojan or get his hands on the mobile device and create a backup to analyse.
```
Comment (kunwaratulhax0r) [inline 2023-risks/m4-inadequate-privacy-controls.md:22]: An attacker can eavesdrop

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+Risks of privacy violations increase due to careless handling of PII by developers. PII should always be processed with the posibility in mind that an attacker could access communication and storage media. Hence, PII should not be stored or transfered
+unless absolutely necessary. If it is to store or transfer, one should consider encrypting or anonymizing it if feasible. Leakage of PII to uncontrollable sinks, e.g., logs, clipboard, url-parameters, must be prevented.
+
+
+# Technical Impacts	
+
+**Impact LOW**
+
+Privacy violations usually have little technical impact on the system as a whole. Only if the PII includes information like authentication data, it can affect traceability. If user data is manipulated it might render the system unusable for that user.
```
Comment (3asm) [inline 2023-risks/m4-inadequate-privacy-controls.md:37]: +1.

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+
+# Business Impacts
+	
+**Impact SEVERE** 
+
+The business impact of privacy violations will typically result in the following at a minimum:
```
Comment (3asm) [inline 2023-risks/m4-inadequate-privacy-controls.md:44]: +1

Regulations are the biggest issue when talking about privacy controls. GDPR, CCPA (California, US), PDPA (Singapore), PIPEDA (Canada), LGPD (Brazil), Data Protection Act 2018 (UK), POPIA (South Africa), PDPL (China) are examples of relevant regulations with known sanctions against companies for not protecting their users' data.

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+- Loss or theft of personally identifiable informationInformation 
+- Reputational damage
+- Financial damage if jurisdiction mandates protection of PII, e.g., the European General Data Protection Regulation
+
+
+# Am I Vulnerable To 'Inadequate Privacy Controls'?
+
+Are you using PII?
```
Comment (3asm) [inline 2023-risks/m4-inadequate-privacy-controls.md:53]: +1.

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+Do you store and transfer them securely (cf. M2, M8)?
+Are protected backups enabled?
+Do they leak to externally accessible sinks, like clipboard, url-parameters, intents/broadcasts, logs/error messages?
+
+
+# How Do I Prevent 'Inadequate Privacy Controls'?
+
+Be aware of the most valuable PII assets you are using in your app. Use threat modeling to assess the most likely ways that privacy violations may occur for your app. Use static analyses to reveal common pitfalls, like logging of PII or leakage to URL query parameters. Consider platform or 3rd party tools to sanitize logs and error messages.
```
Comment (3asm) [inline 2023-risks/m4-inadequate-privacy-controls.md:61]: Not sure how relevant. 3rd party SDK and analytics data leakage are not covered. I would also argue that Static Analysis is not efficient enough to detect leaks of PII.

File: 2023-risks/m4-inadequate-privacy-controls.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+Typical sources for PII are either protected or hard to access, e.g., the sandbox of the app, the network communication with the server, the app's logs and backups. Some are easier to observe, like URL query parameters and clipboard content.
+
+Obtaining PII thus requires the attacker to breach security on another level. He could eavesdrop on the network communication, access file system, clipboard or logs with a trojan or get his hands on the mobile device and create a backup to analyse.
+
+# Security Weakness	
+
+**Prevalence COMMON** <br/>
+**Detectability EASY**
+
+Risks of privacy violations increase due to careless handling of PII by developers. PII should always be processed with the posibility in mind that an attacker could access communication and storage media. Hence, PII should not be stored or transfered
```
Comment (kunwaratulhax0r) [inline 2023-risks/m4-inadequate-privacy-controls.md:29]: Risks of privacy violations increase due to careless handling of PII by developers. PII should always be processed with the possibility in mind that an attacker could access communication and storage media. Hence, PII should not be stored or transferred
unless absolutely necessary. If it is to store or transfer, one should consider encrypting or anonymizing it if feasible. Leakage of PII to uncontrollable sinks, e.g., logs, clipboard, URL-parameters, must be prevented.

An attacker could manipulate or abuse the data collected by the mobile app, potentially leading to privacy infringements or identity theft.

Comment (3asm) [inline 2023-risks/m4-inadequate-privacy-controls.md:29]: This mentions careless handling. Should (malicious) overcollection of PII also be included?

**Review by kunwaratulhax0r — CHANGES_REQUESTED** [review]
Added comments.

**Pull request closed without merging**